src/picologging/formatter.cxx

```diff
@@ -170,17 +170,22 @@ PyObject* Formatter_format(Formatter *self, PyObject *record){
             }
             PyUnicode_Append(&result, logRecord->excText);
         }
-        if (logRecord->stackInfo != Py_None && logRecord->stackInfo != Py_False){
-            if (!PYUNICODE_ENDSWITH(result, self->_const_line_break)){
-                PyUnicode_Append(&result, self->_const_line_break);
-            }
-            if (PyUnicode_Check(logRecord->stackInfo)){
-                PyUnicode_Append(&result, logRecord->stackInfo);
+        if (logRecord->stackInfo != Py_None && logRecord->stackInfo != Py_False ) {
+            if (PyUnicode_Check(logRecord->stackInfo) ) {
+                if (PyUnicode_GET_LENGTH(logRecord->stackInfo) > 0) {
```

Comment:
I only put the check for empty string here, but I think that's actually equivalent functionally to the CPython code:

```
        if record.stack_info:
            if s[-1:] != "\n":
                s = s + "\n"
            s = s + self.formatStack(record.stack_info)
```

In CPython, if the stack_info object isn't a string but some other object, then that check would be truthy, and the newline would get added without checking if the stringification of the stack_info object would be an empty string.